rfc.md

```diff
@@ -0,0 +1,215 @@
+      phenotypes */
+    alleles: "a-a b-b ..."
+    /** optional. Explicitly specifying the authored syntax */
+    authoredAlleles: "a-a b- -c d-[D1, D2] e-^[E1] XY"
+    /** optional */
+    sex: "male"
+  }): Organism
+
+  breed(org: Organism, org: Organism, crossover=true): Organism
+
+  breed(org: Organism, org: Organism, quantity: number, crossover=true): Organism[]
+
+  fertilize(gamete: Gamete, gamete: Gamete): Organism
+```
+
+## Problem: Species
+
+In biologica 1.0, we need to pass in a BioLogica.Species object in any Organism creation
+method. This seems ugly, and would be even more so now because we'd no longer create
+Organism classes that have their own references to the species, so we'd have to *either*
+add the entire species spec to every organism, *or* pass in the species spec in every
+single method (e.g. `breed(Organism, Organism, Species)`).
+
+One solution would be to have an initialization method that sets up the species from the start:
+
+```
+  import biologica from `biologica`
+
+  const Drakes = biologica(drakeSpecies)
+  const org1 = Drakes.organism(...)
+  const child = Drakes.breed(org1, org2)
+
+  // instead of
+
+  import { organism, breed } from `biologica`
+
+  const org1 = organism(..., drakeSpecies)
+  const child = breed(org1, org2, drakeSpecies)
+```
+The top example seems a little uglier than the alternative below, but saves us from
+including `drakeSpecies` everywhere. Thoughts?
+
+Edit: Or maybe we can keep the top solution but make it cleaner by using
+
+```
+  import biologica from `biologica`
+  const { organism, breed } = biologica(drakeSpecies)
+
+  const org1 = organism(...)
+  const child = breed(org1, org2)
+```
+
+## Small problem of not using classes
+
+This is probably not so important, but classes with methods can often be read in a
+more literate manner: If we wanted to, say, get the image src for an authored organism
+in 1.0, we could write
+
+`new Organism(authoredAlleles).getImageSrc()`
+
+In 2.0 it seems that we would either need to split it into two lines or use the backwards
+
+`getImageSrc(organism(authoredAlleles))`
+
+(or we could use the [futuristic](https://github.com/jussi-kalliokoski/trine/blob/5b735cbfb6b28ae94bac0446d9ecd5ce51fb149b/README.md) syntax `organism(authoredAlleles)::getImageSrc()` but
+let's not.)
+
+## Authored alleles syntax
+
+```
+  a1-a2 b1- -c2 d1 e1-[e2, e3] f1-^[f3] XY
+```
+
+This makes the new dashed syntax the prime format (though we probably need to suport authoring
+with the old `a:a1,b:a1` synatax) and adds a couple features Geni* has been needing
+
+ * `a1-a2 b1- -c2`  
+   Any allele listed on one side or the other of a dash indicates a requirement for that
+   allele on the left or right chromosome. (Note, we need to make explicit which side is
+  from which parent. This is implicit, not explicit, in 1.0).
+* `d1`  
+  A required allele that could be on either side.
+* `[e2, e3]`  
+  Either e2 or e3
+* `^[f3]`  
+   Not f3
+* `XY`  
```

Comment:
I like including this in the allele string, it also makes it easier to support something like `XXY` down the line if we wanted. In general, I think it is a good idea to include a separate field for `sex` (as you suggested) since it could mean [so many different things](https://medical-dictionary.thefreedictionary.com/phenotypic+sex) - it isn't necessarily redundant.

Reply:
True. And it could actually be part of the species definition how the `sex` string is defined. Most would probably just state `Y chromosome ? "male" : "female"`, but it would leave it open to change.

(Actually, the original Dragons 15 years ago had XX and XY be the opposite that they are now, because that's the way birds are: males are ZZ and females are ZW, where Z is the larger chromosome. That was quickly abandoned, I believe, because everyone agreed that just added too much confusion.)